mypy/stubgen.py

```diff
@@ -1003,6 +1014,10 @@ def visit_assignment_stmt(self, o: AssignmentStmt) -> None:
                 assert isinstance(o.rvalue, CallExpr)
                 self.process_namedtuple(lvalue, o.rvalue)
                 continue
+            if isinstance(lvalue, NameExpr) and self.is_typeddict(o.rvalue):
+                assert isinstance(o.rvalue, CallExpr)
```

Comment:
We could do away with this assertion if we changed the signature of the `is_typeddict` method so that it only accepts `CallExpr` nodes, and did the `if isinstance(node, CallExpr)` check prior to passing the node to `self.is_typeddict()` instead of it being the the first thing we do inside the `is_typeddict` method